narwhals/_arrow/series.py

```diff
@@ -209,6 +209,10 @@ def drop_nulls(self) -> ArrowSeries:
         pc = get_pyarrow_compute()
         return self._from_native_series(pc.drop_null(self._native_series))
 
+    def shift(self, n: int) -> Self:
+        pc = get_pyarrow_compute()
+        return self._from_native_series(pc.shift(self._native_series, n))
```

Comment:
This might end up being not so straightforward after all. We should _manually_ slice the original array and build the new one, this will also be dependent on the sign of `n`